src/debugger/src/cli_debugger.cpp

```diff
@@ -0,0 +1,160 @@
+#include "debugger/cli_debugger.h"
+#include <iostream>
+
+using namespace debugger;
+using namespace std;
+
+CliDebugger::CliDebugger(std::unique_ptr<DebuggerCore> debugger_core)
+    : debugger_core(move(debugger_core)),
+      command_parser("tdb", "The tvp Debugger") {
+
+	// clang-format off
+	command_parser.add_options()
+	    ("run", "Run the ROM till it encounters a breakpoint")
+	    ("step", "Execute one System tick")
+	    ("bp-set", "Set a breakpoint", cxxopts::value<string>())
+	    ("bp-remove", "Remove a breakpoint", cxxopts::value<string>())
+	    ("bp_cycles-set", "Set a CPU cycle breakpoint", cxxopts::value<ClockCycles>()->default_value("0"))
+	    ("bp_cycles-remove", "Remove a breakpoint", cxxopts::value<ClockCycles>()->default_value("0"))
+	    ("bp_ticks-set", "Set a breakpoint", cxxopts::value<ClockCycles>()->default_value("0"))
+	    ("bp_ticks-remove", "Remove a breakpoint", cxxopts::value<ClockCycles>()->default_value("0"))
+	    ("peek", "Peek into the next \'x\' lines of the ROM", cxxopts::value<uint32_t>()->default_value("0"))
+	    ("bp-view", "View All Instruction Breakpoints")
+	    ("bp_ticks-view", "View all Tick Breakpoints")
+	    ("bp_cycles-view", "View all CPU Cycle Breakpoints")
+        ("help", "Print this information");
+
+	// clang-format on
+}
+
+void CliDebugger::tick() {
+	std::string str;
+
+	do {
+		cout << "(tdb) ";
+		getline(std::cin, str);
+
+		//  A hack to make CLI parsing compatible with cxxopts.
+		str = "./tdb --" + str;
+		auto [argc, argv] = cmd_string_to_argv(str);
+
+		status = run_command(argc, argv);
```

Comment:
This can be a local variable.

As a rule - If a variable is not required outside the context of just one function, like `status` here, it can be a local variable instead of a class member.

Reply:
Understood. I will keep this in mind for future PR's.